src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -25,18 +25,61 @@ T min(T a, T b)
 
 namespace precice::cplscheme {
 
+namespace {
+bool compatibleTimeWindowSizes(const CouplingScheme &impl, const CouplingScheme &expl)
+{
+  if (!impl.hasTimeWindowSize() || !expl.hasTimeWindowSize()) {
+    return true;
+  }
+  double idt = impl.getTimeWindowSize();
+  double edt = expl.getTimeWindowSize();
+  if (math::equals(edt, idt)) {
+    return true; // equal is compatible
+  }
+  if (math::smaller(edt, idt)) {
+    return false; // smaller edt is incompatible
+  }
+  return math::equals(std::remainder(edt, idt), 0.0);
+}
+} // namespace
+
+void CompositionalCouplingScheme::checkCompatibleTimeWindowSizes(const CouplingScheme &impl, const CouplingScheme &expl) const
+{
+  if (compatibleTimeWindowSizes(impl, expl)) {
+    return;
+  }
+  auto   local        = expl.localParticipant();
+  auto   explPartners = expl.getCouplingPartners();
+  auto   implPartners = impl.getCouplingPartners();
```

Comment:
Why Partners and not Participants? (Not relevant for the release, but maybe a tiny refactoring for `develop`)

Reply:
Possibly because a coupling partner always is a participant that takes part in a scheme.

Reply:
I'm just wondering about the wording here. Is partner and participant the same thing?

Reply:
Not every participant in the simulation needs to be coupled to the local participant.
So, this distinction kind of makes sense. It could also just be `getCoupledParticipants()`.